.github/workflows/ci.yml

```diff
@@ -10,6 +10,20 @@ defaults:
     shell: bash
 
 jobs:
+  defaults:
+    name: Set up defaults
+    runs-on: ubuntu-latest
+    outputs:
+      php: ${{ steps.defaults.outputs.php }}
+      browser: ${{ steps.defaults.outputs.browser }}
+      selenium: ${{ steps.defaults.outputs.selenium }}
+    steps:
+      - id: defaults
+        run: |
+          echo 'php="8.3"' >> "$GITHUB_OUTPUT"
+          echo 'browser="chrome"' >> "$GITHUB_OUTPUT"
+          echo 'selenium="4"' >> "$GITHUB_OUTPUT"
```

Comment:
Can this code be simplified according to https://docs.github.com/en/enterprise-server@3.10/actions/writing-workflows/choosing-what-your-workflow-does/setting-a-default-shell-and-working-directory ?

Maybe move them to a specific job instead of above all jobs.

```
jobs:
  job1:
    runs-on: ubuntu-latest
    defaults:
      run:
        shell: bash
        working-directory: ./scripts
```

Then use these defaults as `{{ jobs.<job_id>.defaults.run }}`.

The `$GITHUB_OUTPUT` stuff looks like a workaround to me.

Reply:
It has to run at the job level, because those defaults are used in a matrix of a job.

The problem is to use variables in a matrix.. apparently it's not possible.

Reply:
As with tests I prefer to do some copy/paste to increase code clarity. If we need to change the default browser/selenium version we can do several `Find & Replace`.

Reply:
The problem with that is that it doesn't work well with short values like '4' or '8.3' (on top of which there are several places that shouldn't be replaced).

Other than that, the code is actually relatively readable (except that first step, but let's be honest, github workflows aren't that readable in general).

Reply:
I see. I've just found https://partial.solutions/2023/advanced-matrix-testing-with-github-actions.html and the example in there looks very similar to your approach.

Let's keep the current implementation in this PR then.